core/database/schema/query.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package schema
+
+import (
+	"context"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/juju/errors"
+)
+
+const schemaTable = `
+CREATE TABLE IF NOT EXISTS schema (
+    id         	INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    version    	INTEGER NOT NULL,
+	hash	   	TEXT NOT NULL,
+    updated_at 	DATETIME NOT NULL,
+    UNIQUE (version)
```

Comment:
Prefer creating a unique index rather than a constrain, with a name conforming to our convention (idx_...).